src/libraries/Common/src/System/Net/Security/MD4.cs

```diff
@@ -66,9 +66,10 @@ internal static void HashData(ReadOnlySpan<byte> source, Span<byte> destination)
 
             Span<byte> buffer = stackalloc byte[64];
             buffer.Clear();
+
             // Initialize the context
-            Span<uint> state = stackalloc uint[4] { 0x67452301, 0xefcdab89, 0x98badcfe, 0x10325476 };
-            Span<uint> count = stackalloc uint[2] { 0, 0 };
+            Span<uint> state = [0x67452301, 0xefcdab89, 0x98badcfe, 0x10325476];
```

Comment:
Is this universally a good idea? If I do this against .NET Framework, this turns a `stackalloc` in to a heap alloc since it is a writable Span.

https://sharplab.io/#v2:EYLgdgpgLgZgHgHwAICYCMBYAUEgzAAlXwGF8BvbfKwgpAFnwFkAKASnMuq4GUAHAQzAAeAK4BLMFAB8+AMb4AvPgDaABjgA2AOx0ArClyq0AGnzqIMWQBN+wABwBOU+od3g/K7JgRncNKtwUXTotDQBdAG5OLioAOWZZViisLgBfaPwMvEIGeL5BUQlpOVYMihTqdKxUoA=

Reply:
Is this code used in pre-.NET 8 builds? I didn't think it was.

In general, though, you're correct that the current code gen for this will allocate. It'd be good if that were changed to stackalloc for primitive types, at least when not inside of a loop. cc: @CyrusNajmabadi, @RikkiGibson, @cston

Reply:
You're right, this one is only .NET 8+.